src/xmipp/libraries/tomo/tomo_detect_landmarks.cpp

```diff
@@ -0,0 +1,1459 @@
+	MultidimArray<double> imageTmp;
+	MultidimArray<double> imageOut;
+	imageOut.initZeros(ySize_d, xSize_d);
+
+	double angleStep = PI / numberFTdirOfDirections;
+
+	for (size_t n = 0; n < numberFTdirOfDirections; n++)
+	{
+	 	imageTmp = image;
+
+		#ifdef DEBUG_DIRECTIONAL_FOURIER
+		std::cout << "xdir= " << cos(n*angleStep) << ", ydir=" << sin(n*angleStep) << std::endl;
+		#endif
+
+		directionalFilterFourier(imageTmp, cos(n*angleStep), sin(n*angleStep));
+
+		imageOut = imageOut + imageTmp;
+	}
+
+	image = image * imageOut;
+}
+
+
+void ProgTomoDetectLandmarks::directionalFilterFourier(MultidimArray<double> &image, double xdir, double ydir) 
+{
+    MultidimArray<std::complex<double>> fftImg;
+	FourierTransformer transformer1;
+
+    transformer1.FourierTransform(image, fftImg, false);
+
+	auto YdimFT=(int)YSIZE(fftImg);
+	auto XdimFT=(int)XSIZE(fftImg);
+
+	Matrix1D<double> freq_fourier_x;
+	Matrix1D<double> freq_fourier_y;
+	MultidimArray< double > freqMap;
+
+	// Initializing the frequency vectors
+	freq_fourier_x.initZeros(XSIZE(fftImg));
+	freq_fourier_y.initZeros(YSIZE(fftImg));
+
+	double u;  // frequency
+
+	// Defining frequency components. First element should be 0, it is set as the smallest number to avoid singularities
+	VEC_ELEM(freq_fourier_y,0) = std::numeric_limits<double>::min();
+	for(size_t k=1; k<YdimFT; ++k){
+		FFT_IDX2DIGFREQ(k,ySize_d, u);
+		VEC_ELEM(freq_fourier_y, k) = u;
+	}
+
+	VEC_ELEM(freq_fourier_x,0) = std::numeric_limits<double>::min();
+	for(size_t k=1; k<XdimFT; ++k){
+		FFT_IDX2DIGFREQ(k,xSize_d, u);
+		VEC_ELEM(freq_fourier_x, k) = u;
+	}
+
+	//Initializing map with frequencies
+	freqMap.resizeNoCopy(fftImg);
+	freqMap.initConstant(1.9);  //Nyquist is 2, we take 1.9 greater than Nyquist
+
+	// Directional frequencies along each direction
+	double uy, ux, uy2;
+	long n=0;
+	for(size_t i=0; i<YdimFT; ++i)
+	{
+		uy = VEC_ELEM(freq_fourier_y, i);
+		uy2 = uy*uy;
+
+		for(size_t j=0; j<XdimFT; ++j)
+		{
+			ux = VEC_ELEM(freq_fourier_x, j);
+			ux = sqrt(uy2 + ux*ux);
+
+			if	(ux<=0.5)
+			{
+				DIRECT_MULTIDIM_ELEM(freqMap,n) = 1/ux;
+				if ((j == 0) && (uy<0))
+				{
+					DIRECT_MULTIDIM_ELEM(freqMap,n) = 1.9;
+				}
+				if ((i == 0) && (j == 0))
+				{
+					DIRECT_MULTIDIM_ELEM(freqMap,n) = 1.9;
+				}
+					
+			}				
+			++n;
+		}
+	}
+	#ifdef DEBUG_OUTPUT_FILES
+	Image<double> img;
+	img() = freqMap;
+	img.write("freqMap.mrc");
+	#endif
+
+	double cosAngle = 0.9848; // 10º cosine
+	auto aux = (8.0/((cosAngle -1)*(cosAngle -1)));
+
+	double lowerBound = 1/targetFS-0.1;
+	double upperBound = 1/targetFS+0.1;
+
+	n = 0;
+	for (int i=0; i<YdimFT; i++)
+	{
+		double uy = VEC_ELEM(freq_fourier_y, i);
+		for (int j=0; j<XdimFT; j++)
+		{
+			double ux = VEC_ELEM(freq_fourier_x, j);
+
+			double iun = DIRECT_MULTIDIM_ELEM(freqMap,n);
+
+
+			if (1/iun<upperBound && 1/iun>lowerBound)
+			{
+				// Si nos queremos y nos amamos porque no nos besamos por donde meamos
```

Comment:
"""""""I did not see this""""""", but @Vilax requests removing it

Reply:
Disappointing... xmipp misses cutting edge techonology here.

Removed